funnel/views/api/whatsapp_events.py

```diff
@@ -0,0 +1,92 @@
+"""Callback handlers from WhatsApp."""
+
+from __future__ import annotations
+
+from flask import abort, current_app, request
+
+from baseframe import statsd
+
+from ... import app
+from ...models import PhoneNumber, PhoneNumberInvalidError, db, sa
+from ...typing import ReturnView
+from ...utils import abort_null
+
+
+@app.route('/api/1/whatsapp/meta_event', methods=['GET'])
+def process_whatsapp_webhook_verification():
+    """Meta requires to verify the webhook URL by sending a GET request with a token."""
+    verify_token = app.config['WHATSAPP_WEBHOOK_SECRET']
+    mode = request.args.get('hub.mode')
+    token = request.args.get('hub.verify_token')
+    challenge = request.args.get('hub.challenge')
+
+    if mode and token:
+        if mode == 'subscribe' and token == verify_token:
+            return challenge, 200
+        return 'Forbidden', 403
+    return 'Success', 200
+
+
+@app.route('/api/1/whatsapp/meta_event', methods=['POST'])
+def process_whatsapp_event() -> ReturnView:
+    """Process WhatsApp callback event from Meta Cloud API."""
+    # Register the fact that we got a WhatsApp event.
+    # If there are too many rejects, then most likely a hack attempt.
+
+    # FIXME: Where is the call verification?
```

Comment:
To be fixed.

Reply:
The verification happens only once during registration of the webhook in the Meta Business Suite

Reply:
This still needs call verification. Anybody can call this endpoint.